bftengine/src/bftengine/messages/ReqMissingDataMsg.hpp

```diff
@@ -18,57 +18,69 @@ namespace impl {
 
 class ReqMissingDataMsg : public MessageBase {
  public:
-  ReqMissingDataMsg(ReplicaId senderId, ViewNum v, SeqNum s);
+  ReqMissingDataMsg(ReplicaId senderId, ViewNum v, SeqNum s, const std::string spanContext = "");
 
   ViewNum viewNumber() const { return b()->viewNum; }
 
   SeqNum seqNumber() const { return b()->seqNum; }
 
-  bool getPrePrepareIsMissing() const { return (b()->flags & 0x2) != 0; }
-  bool getPartialProofIsMissing() const { return (b()->flags & 0x4) != 0; }
-  bool getPartialPrepareIsMissing() const { return (b()->flags & 0x8) != 0; }
-  bool getPartialCommitIsMissing() const { return (b()->flags & 0x10) != 0; }
-  bool getFullCommitProofIsMissing() const { return (b()->flags & 0x20) != 0; }
-  bool getFullPrepareIsMissing() const { return (b()->flags & 0x40) != 0; }
-  bool getFullCommitIsMissing() const { return (b()->flags & 0x80) != 0; }
-  bool getSlowPathHasStarted() const { return (b()->flags & 0x100) != 0; }
+  bool getPrePrepareIsMissing() const { return b()->flags.bits.prePrepareIsMissing != 0; }
+  bool getPartialProofIsMissing() const { return b()->flags.bits.partialProofIsMissing != 0; }
+  bool getPartialPrepareIsMissing() const { return b()->flags.bits.partialPrepareIsMissing != 0; }
+  bool getPartialCommitIsMissing() const { return b()->flags.bits.partialCommitIsMissing != 0; }
+  bool getFullCommitProofIsMissing() const { return b()->flags.bits.fullCommitProofIsMissing != 0; }
+  bool getFullPrepareIsMissing() const { return b()->flags.bits.fullPrepareIsMissing != 0; }
+  bool getFullCommitIsMissing() const { return b()->flags.bits.fullCommitIsMissing != 0; }
+  bool getSlowPathHasStarted() const { return b()->flags.bits.slowPathHasStarted != 0; }
 
-  uint16_t getFlags() const { return b()->flags; }
+  uint16_t getFlags() const { return b()->flags.flags; }
 
   void resetFlags();
 
-  void setPrePrepareIsMissing() { b()->flags |= 0x2; }
-  void setPartialProofIsMissing() { b()->flags |= 0x4; }
-  void setPartialPrepareIsMissing() { b()->flags |= 0x8; }
-  void setPartialCommitIsMissing() { b()->flags |= 0x10; }
-  void setFullCommitProofIsMissing() { b()->flags |= 0x20; }
-  void setFullPrepareIsMissing() { b()->flags |= 0x40; }
-  void setFullCommitIsMissing() { b()->flags |= 0x80; }
-  void setSlowPathHasStarted() { b()->flags |= 0x100; }
+  void setPrePrepareIsMissing() { b()->flags.bits.prePrepareIsMissing = 1; }
+  void setPartialProofIsMissing() { b()->flags.bits.partialProofIsMissing = 1; }
+  void setPartialPrepareIsMissing() { b()->flags.bits.partialPrepareIsMissing = 1; }
+  void setPartialCommitIsMissing() { b()->flags.bits.partialCommitIsMissing = 1; }
+  void setFullCommitProofIsMissing() { b()->flags.bits.fullCommitProofIsMissing = 1; }
+  void setFullPrepareIsMissing() { b()->flags.bits.fullPrepareIsMissing = 1; }
+  void setFullCommitIsMissing() { b()->flags.bits.fullCommitIsMissing = 1; }
+  void setSlowPathHasStarted() { b()->flags.bits.slowPathHasStarted = 1; }
```

Comment:
this is cool indeed, but it's absolutely unrelated to the span context changes... Such improvements should be done in separate, more targetted PR-s

Reply:
Agreed, just wanted to add a small improvement on the way...